embulk-core/src/main/java/org/embulk/spi/util/dynamic/LongColumnSetter.java

```diff
@@ -32,8 +30,15 @@ public void set(long v) {
     public void set(double v) {
         long lv;
         try {
-            // TODO configurable rounding mode
-            lv = DoubleMath.roundToLong(v, RoundingMode.HALF_UP);
+            final double roundedDouble = Math.rint(v);
+            final double diff = v - roundedDouble;
+            if (diff == 0.5) {
+                lv = (long) (v + 0.5);
+            } else if (diff == -0.5) {
+                lv = (long) (v - 0.5);
+            } else {
+                lv = (long) roundedDouble;
+            }
```

Comment:
I think we have an issue if v has an even integer part, e.g 2.5.

Reply:
What's the issue?  If `v` is exact `2.5`,

* `roundedDouble` would be `2.0` from the definition of `Math.rint()`.
* `diff` would be `0.5` from `2.5 - 2.0`.
* `lv` would be `3` from `2.5 + 0.5`.

Then it's the expected `HALF_UP` behavior.

Reply:
Sorry, I just have a mistake. Actually, the issue happens when v has an integer part is 1.5

`v` is exact `1.5`
rounded double would be `2.0`
`diff` would be `-0.5` from `1.5 - 2.0`
lv would be `1` from `1.5 -0 .5`.
I think we expect 2 in here.

Reply:
Ah, probably you're right. I'll have another look into Guava, and have a comparison.

Reply:
Thanks for your good catch. I should have used the sign of `v`, not the sign of `diff`, for `0.5`. Fixed it in: e65f6ecd21a728df893bee57dc47fc691f13470e